internal/common/rules/basicruleevaluator_test.go

```diff
@@ -43,12 +54,38 @@ func registerNewRealValuedRule(t *testing.T) {
 
 	v := []float64{1, 0, 0, 0, -4, 0, -1, -1, 0, 2, 0, 0, 0, 0, 2, 0, 0, 1, 0, -1}
 	CoreMatrix := mat.NewDense(4, 5, v)
-	aux := []float64{1, 1, 3, 0}
+	aux := []float64{1, 1, 4, 0}
 	AuxiliaryVector := mat.NewVecDense(4, aux)
 
-	_, ruleError := RegisterNewRule(name, reqVar, *CoreMatrix, *AuxiliaryVector, false)
+	_, ruleError := RegisterNewRule(name, reqVar, *CoreMatrix, *AuxiliaryVector, false, RuleLink{
+		Linked: false,
+	})
 	if ruleError != nil {
 		t.Errorf("Problem with registering new real valued rule in test, error message : '%v'", ruleError.Error())
 	}
 	// Check internal/clients/team3/client.go for an implementation of a basic evaluator for this rule
 }
+
+func registerNewLinkedRule(t *testing.T) {
+	name := "Linked test rule"
+	reqVar := []VariableFieldName{
+		NumberOfIslandsContributingToCommonPool,
+		NumberOfFailedForages,
+		NumberOfBrokenAgreements,
+		MaxSeverityOfSanctions,
+	}
+
+	v := []float64{1, 0, 0, 0, -4, 0, -1, -1, 0, 2, 0, 0, 0, 0, 2, 0, 0, 1, 0, -1}
+	CoreMatrix := mat.NewDense(4, 5, v)
+	aux := []float64{1, 1, 3, 0}
+	AuxiliaryVector := mat.NewVecDense(4, aux)
+
+	_, ruleError := RegisterNewRule(name, reqVar, *CoreMatrix, *AuxiliaryVector, false, RuleLink{
+		Linked:     true,
+		LinkType:   ParentFailAutoRulePass,
+		LinkedRule: "Kinda Complicated Rule",
```

Comment:
Where is this rule defined

Reply:
BasicRuleEvaluator feeds off whatever is in the AvailableRulesCache, so this rule is a default entry in there, it was a convenient one to use ;)